straxen/storage/rundb.py

```diff
@@ -235,18 +239,27 @@ def find_several(self, keys: typing.List[strax.DataKey], **kwargs):
         if self.runid_field == 'name':
             run_query = {'name': {'$in': [key.run_id for key in keys]}}
         else:
-            run_query = {f'{self.runid_field}': {'$in': [int(key.run_id) for key in keys if not key.run_id.startswith('_')]}}
+            run_query = {f'{self.runid_field}': {
+                '$in': [int(key.run_id) for key in keys if not key.run_id.startswith('_')]}}
         dq = self._data_query(keys[0])
 
         # dict.copy is sometimes not sufficient for nested dictionary
         projection = deepcopy(dq)
         projection.update({
             k: True
             for k in f'name number'.split()})
+        results_dict = self._parse_documents(
+            self.collection.find(
+                {**run_query, **dq},
+                projection=projection),
+        )
 
+        return [results_dict.get(k.run_id, False)
+                for k in keys]
+
+    def _parse_documents(self, find: iter):
```

Comment:
I would rename find --> docs/documents but that's just personal taste.

Reply:
Thanks, as a reviewer, you are certainly entitled to expressing your personal tastes 👍 I think you are right.